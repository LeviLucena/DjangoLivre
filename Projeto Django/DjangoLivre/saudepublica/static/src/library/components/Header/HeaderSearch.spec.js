import { shallowMount } from '@vue/test-utils'
import BrHeader from './Header.ce.vue'
import BrHeaderAction from './HeaderAction.ce.vue'
import BrHeaderActionSearch from './HeaderActionSearch.ce.vue'
import BrHeaderSearch from './HeaderSearch.ce.vue'

describe('Header - Search', () => {
  test('it renders BrHeadersearch', async () => {
    const wrapper = shallowMount(BrHeader, {
      propsData: {
        hasSearch: true,
      },
    })
    expect(wrapper.find('.header-search').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('it calls "closeInputSearch" method', async () => {
    const spyCloseInputSearch = jest.spyOn(BrHeaderSearch.methods, 'closeInputSearch')
    const wrapper = shallowMount(BrHeader, {
      props: {
        hasSearch: true,
        compact: true,
      },
      slots: {
        headerAction: `
        <div slot="headerAction">
          <div >
            <br-header-action>
            </br-header-action>
          </div>
        </div>
        `,
      },
      global: {
        stubs: {
          'br-header-action': BrHeaderAction,
          'br-header-search': BrHeaderSearch,
          'br-header-action-search': BrHeaderActionSearch,
        },
      },
    })
    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('[data-target=".header-search"]').trigger('click')
    wrapper.vm.$nextTick(() => {
      expect(wrapper.element).toMatchSnapshot()
      expect(wrapper.find('.header-search.active').exists()).toBe(true)
      wrapper.find('[data-dismiss="search"]').trigger('click')
      expect(spyCloseInputSearch).toHaveBeenCalled()
    })
  })
})
