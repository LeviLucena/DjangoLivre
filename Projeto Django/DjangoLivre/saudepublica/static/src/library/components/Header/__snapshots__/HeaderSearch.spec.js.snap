// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header - Search it calls "closeInputSearch" method 1`] = `
<header
  class="br-header mb-4 compact"
>
  <div
    class="container-lg"
  >
    <div
      class="header-top"
    >
      <div
        class="header-logo"
      >
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      <div
        slot="headerAction"
      >
        <div>
          <div
            class="header-actions"
            style="float: right;"
          >
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
            <div
              class="header-search-trigger"
              style="display: block;"
            >
              <br-button-stub
                active="false"
                aria-label="Abrir Busca"
                block="false"
                circle="true"
                data-target=".header-search"
                data-toggle="search"
                density="middle"
                disabled="false"
                icon="search"
                inverted="false"
                loading="false"
              />
            </div>
            <!--v-if-->
          </div>
        </div>
      </div>
      
    </div>
    <div
      class="header-bottom"
    >
      <div
        class="header-menu"
      >
        <!--v-if-->
        <div
          class="header-info"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div
        class="header-search"
      >
        <form>
          <br-input-stub
            autofocus="false"
            icon="search"
            iconsubmit="true"
            id="searchbox4"
            ishighlight="true"
            ispassword="false"
            label="Texto da pesquisa"
            mask=""
            newclass="has-icon"
            placeholder="O que você procura?"
          />
          <br-button-stub
            active="false"
            aria-label="Fechar Busca"
            block="false"
            circle="true"
            class="search-close ml-1"
            data-dismiss="search"
            density="middle"
            disabled="false"
            inverted="false"
            loading="false"
          />
        </form>
      </div>
    </div>
  </div>
</header>
`;

exports[`Header - Search it calls "closeInputSearch" method 2`] = `
<header
  class="br-header mb-4 compact"
>
  <div
    class="container-lg"
  >
    <div
      class="header-top"
    >
      <div
        class="header-logo"
      >
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      <div
        slot="headerAction"
      >
        <div>
          <div
            class="header-actions"
            style="float: right;"
          >
            <!--v-if-->
            <!--v-if-->
            <!--v-if-->
            <div
              class="header-search-trigger"
              style="display: block;"
            >
              <br-button-stub
                active="false"
                aria-label="Abrir Busca"
                block="false"
                circle="true"
                data-target=".header-search"
                data-toggle="search"
                density="middle"
                disabled="false"
                icon="search"
                inverted="false"
                loading="false"
              />
            </div>
            <!--v-if-->
          </div>
        </div>
      </div>
      
    </div>
    <div
      class="header-bottom"
    >
      <div
        class="header-menu"
      >
        <!--v-if-->
        <div
          class="header-info"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div
        class="header-search active"
      >
        <form>
          <br-input-stub
            autofocus="false"
            icon="search"
            iconsubmit="true"
            id="searchbox4"
            ishighlight="true"
            ispassword="false"
            label="Texto da pesquisa"
            mask=""
            newclass="has-icon"
            placeholder="O que você procura?"
          />
          <br-button-stub
            active="false"
            aria-label="Fechar Busca"
            block="false"
            circle="true"
            class="search-close ml-1"
            data-dismiss="search"
            density="middle"
            disabled="false"
            inverted="false"
            loading="false"
          />
        </form>
      </div>
    </div>
  </div>
</header>
`;

exports[`Header - Search it renders BrHeadersearch 1`] = `
<header
  class="br-header mb-4"
>
  <div
    class="container-lg"
  >
    <div
      class="header-top"
    >
      <div
        class="header-logo"
      >
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      
    </div>
    <div
      class="header-bottom"
    >
      <div
        class="header-menu"
      >
        <!--v-if-->
        <div
          class="header-info"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div
        class="header-search"
      >
        <br-header-search-stub
          id="searchbox1"
          label="Texto da pesquisa"
          placeholder="O que você procura?"
        />
      </div>
    </div>
  </div>
</header>
`;
